kernel/schemes/file.rs

```diff
@@ -41,22 +41,18 @@ impl Resource for FileResource {
         })
     }
 
-    fn path(&self, buf: &mut [u8]) -> Result <usize> {
-        let mut i = 0;
-
+    fn path(&self, buf: &mut [u8]) -> Result<usize> {
         let path_a = b"file:/";
```

Comment:
You could get an iter_mut() at the top of this path function, then use it without having to skip